### ISSUE TITLE
e2e_subs:  Automate artifact upload to reduce Indexer make e2e drift

### ISSUE THREAD

**Issue by AlgoStephenAkiki** [algorand/indexer#940]
@michaeldiamant commented on [Mon Mar 07 2022](https://github.com/algorand/go-algorand-internal/issues/1933)

## Problem

Indexer `make e2e` relies on a manual process to update its test input. 
 Drift occurs between go-algorand `e2e_subs` and Indexer `make e2e` test input.  Consequently, go-algorand maintainers miss Indexer incompatibilities.

Notes:
* For a process review, see https://github.com/algorand/indexer/blob/develop/misc/README.md.
* _Disclaimer:_  I do _not_ consider myself an Indexer authority.  Consider including a Lamprey team member during discussion. 

## Solution

Incremental improvement opportunity:  Automate go-algorand `e2e_subs` output upload on merge to `master`.

The proposal brings these benefits:
* Eliminates _most_ drift.
* For go-algorand maintainers, removes artifact upload from mental consideration. 
* Reduces the number of individuals with permissive AWS access. 

### Requirements
* For indexer maintainers, preserve a way to manually specify `make e2e` test input source.
  * Scenarios will continue to arise where builds must be unblocked by pointing to a specific test input.
* Since the proposal implies uploading artifacts (much) more frequently, implement a retention policy.  
  * If purely time-based, ensure at least 1 artifact remains present at all times.
* For debugging, preserve an easy means to correlate the `make e2e` input artifact with the artifact's source (e.g. go-algorand PR that generated the artifact).  
* Update docs to reflect thew new process.
  * https://github.com/algorand/indexer/blob/develop/misc/README.md
  * https://github.com/algorand/go-algorand/blob/master/test/README.md  

### Implementation details
* Ideally, eliminate duplicative S3 bucket and object prefixes.  Keeping these prefixes in-sync is error prone.  Obviously, a manual override capability must be preserved.  Duplication sources:
  * https://github.com/algorand/indexer/blob/develop/misc/e2elive.py
  * https://github.com/algorand/indexer/blob/develop/misc/buildtestdata.sh 

### Out of scope:
* In principle, `e2e_sub` changes to `rel/*` branches ought to be tested too.  However, I suggest focusing _only_ on `master`.   
  * I think it'll be too complex to correlate artifacts across repos _and_ branches. 
  * Additionally, I think tackling `rel/*` is better served by a wholesale different approach.

## Dependencies

N/A

## Urgency

Unclear
